python/src/iceberg/io/base.py

```diff
@@ -142,12 +158,25 @@ class FileIO(ABC):
 
     @abstractmethod
     def new_input(self, location: str) -> InputFile:
-        """Get an InputFile instance to read bytes from the file at the given location"""
+        """Get an InputFile instance to read bytes from the file at the given location
+
+        Args:
+            location(str): The URI to the file
+        """
 
     @abstractmethod
     def new_output(self, location: str) -> OutputFile:
-        """Get an OutputFile instance to write bytes to the file at the given location"""
+        """Get an OutputFile instance to write bytes to the file at the given location
+
+        Args:
+            location(str): The URI to the file
```

Comment:
small nit, this isn't always a URI is it?  For local file systems it is just a file path which I don't think

Reply:
Updated! If a path with no scheme is provided, it's assumed as a path to a local file. Updated the tests as well and it feels much better not having to prepend `file:` :D